apps/src/applab/api-entry.js

```diff
@@ -1,28 +1,138 @@
+  "serverLevelId":2176,
+  "gameDisplayName":"App Lab",
+  "publicCaching":false,
+  "is13Plus":true,
+  "hasContainedLevels":false,
+  "hideSource":true,
+  "share":true,
+  "labUserId":"x+OhD4/hmGgtPrHVlrC32TFHAdo",
+  "firebaseName":"cdo-v3-dev",
+  "firebaseAuthToken":"eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ2IjowLCJpYXQiOjE0ODM5OTg4MjksImQiOnsidWlkIjoiOTIiLCJpc19kYXNoYm9hcmRfdXNlciI6dHJ1ZX19.DX8PP0Q8EDGg7UtMbhT2sT-h39LvDsuuPbA6YesXLG8",
+  "firebaseChannelIdSuffix":"-DEVELOPMENT-pcardune",
+  "isSignedIn":true,
+  "pinWorkspaceToBottom":true,
+  "hasVerticalScrollbars":true,
+  "showExampleTestButtons":true,
+  "rackEnv":"development",
+  "report":{
+    "fallback_response":null,
+            "callback":null,
+            "sublevelCallback":null},
+  "sendToPhone":true,
+  "send_to_phone_url":"http://localhost-studio.code.org:3000/sms/send",
+  "copyrightStrings":{
+    "thank_you":"We%20thank%20our%20%3Ca%20href=%22https://code.org/about/donors%22%3Edonors%3C/a%3E,%20%3Ca%20href=%22https://code.org/about/partners%22%3Epartners%3C/a%3E,%20our%20%3Ca%20href=%22https://code.org/about/team%22%3Eextended%20team%3C/a%3E,%20our%20video%20cast,%20and%20our%20%3Ca%20href=%22https://code.org/about/advisors%22%3Eeducation%20advisors%3C/a%3E%20for%20their%20support%20in%20creating%20Code%20Studio.",
+    "help_from_html":"We especially want to recognize the engineers from Google, Microsoft, Facebook, Twitter, and many others who helped create these materials.",
+    "art_from_html":"Minecraft\u0026%238482;%20\u0026copy;%202017%20Microsoft.%20All%20Rights%20Reserved.%3Cbr%20/%3EStar%20Wars\u0026%238482;%20\u0026copy;%202017%20Disney%20and%20Lucasfilm.%20All%20Rights%20Reserved.%3Cbr%20/%3EFrozen\u0026%238482;%20\u0026copy;%202017%20Disney.%20All%20Rights%20Reserved.%3Cbr%20/%3EIce%20Age\u0026%238482;%20\u0026copy;%202017%2020th%20Century%20Fox.%20All%20Rights%20Reserved.%3Cbr%20/%3EAngry%20Birds\u0026%238482;%20\u0026copy;%202009-2017%20Rovio%20Entertainment%20Ltd.%20All%20Rights%20Reserved.%3Cbr%20/%3EPlants%20vs.%20Zombies\u0026%238482;%20\u0026copy;%202017%20Electronic%20Arts%20Inc.%20All%20Rights%20Reserved.%3Cbr%20/%3EThe%20Amazing%20World%20of%20Gumball%20is%20trademark%20and%20\u0026copy;%202017%20Cartoon%20Network.",
+    "code_from_html":"Code.org%20uses%20p5.play,%20which%20is%20licensed%20under%20%3Ca%20href=%22http://www.gnu.org/licenses/old-licenses/lgpl-2.1-standalone.html%22%3Ethe%20GNU%20LGPL%202.1%3C/a%3E.",
+    "powered_by_aws":"Powered by Amazon Web Services",
+    "trademark":"\u0026copy;%20Code.org,%202017.%20Code.org\u0026reg;,%20the%20CODE%20logo%20and%20Hour%20of%20Code\u0026reg;%20are%20trademarks%20of%20Code.org."
+  },
+  "teacherMarkdown":null,
+  "dialog":{
+    "skipSound":false,
+    "preTitle":null,
+    "fallbackResponse":"null",
+    "callback":null,
+    "sublevelCallback":null,
+    "app":"applab",
+    "level":"custom",
+    "shouldShowDialog":true
+  },
+  "locale":"en_us"
 };
 
+Applab.render = function (){};
```

Comment:
Not going to get solved here, but this indicates a need for a better division between App Lab the execution environment (interpreter, commands, student assets, etc) and App Lab the development environment (UI, editors, debugging, save/load, etc).